ecommerce-ui/src/App.test.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {shallow, render, mount} from 'enzyme';
+import App from './App';
+import { configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import Listing from './listings';
+import ShoppingCart from './shoppingCart';
+
+configure({ adapter: new Adapter() });
+
+it('renders without crashing', () => {
+  const div = document.createElement('div');
+  ReactDOM.render(<App />, div);
+  ReactDOM.unmountComponentAtNode(div);
+});
+
+it('shallow renders', () => {
+  const wrapper = shallow(<App />);
+  expect(wrapper).toMatchSnapshot();
+});
+
+it('shallow renders', () => {
+  const wrapper2 = shallow(<Listing />);
```

Comment:
You are getting the   TypeError: Cannot read property 'map' of undefined
because you are rendering a component <Listing /> but not passing it any props to use. when it gets to the renderer and tries to do:  
this.props.listing.map((item, idx) =>
this.props.listing is undefined so it throws an error.